src/lib/io/sol-iio.c

```diff
@@ -88,6 +89,14 @@ struct resolve_absolute_path_data {
     int id;
 };
 
+struct open_list {
+    struct sol_list list;
+    sol_iio_device *device;
+    sol_iio_config *config;
+};
```

Comment:
Besides keeping a list of open devices, you need to refcount uses. Otherwise, first one that closes a device will make it invalid for others.